crates/core/src/metadata_store.rs

```diff
@@ -263,6 +274,21 @@ impl<T: ProvisionedMetadataStore + Sync> MetadataStore for T {
     }
 }
 
+macro_rules! metrics_helper {
+    (@success=$ok:ident, @error=$err:ident, @duration=$dur:ident, $block:block) => {{
+        let start_time = tokio::time::Instant::now();
+        let result = $block;
+        metrics::histogram!($dur).record(start_time.elapsed());
+        if result.is_ok() {
+            metrics::counter!($ok).increment(1);
+        } else {
+            metrics::counter!($err).increment(1);
+        }
+
+        result
+    }};
+}
+
```

Comment:
I'm not sure if it's worth a macro if this is used only a couple of times to be honest. I'm not sure if it _needs_ a macro if it's just the dimension name that will report the status.

Reply:
It's only to not repeat myself. It's easier to make changes for example.